Unguarded platform specific code

In the `FileIo` constructor there is an unguarded call to `StringFromWString`, but `StringFromWString` is only defined in a `_WIN32` guarded section.

When compiling under Linux we get the following error:
> OP2-Landlord/FileIo.cpp: In constructor ‘FileIo::FileIo(SDL_Window&)’:
> OP2-Landlord/FileIo.cpp:57:22: error: ‘StringFromWString’ was not declared in this scope
>    57 |     mPathSeparator = StringFromWString(separatorWchar);
>       |                      ^~~~~~~~~~~~~~~~~

Any platform specific code should be guarded by platform specific include guards. Code that is not guarded should work across all platforms.

--- a/OP2-Landlord/FileIo.cpp
+++ b/OP2-Landlord/FileIo.cpp
@@ -58,7 +58,7 @@ FileIo::FileIo(SDL_Window& window) :
 {
     SDL_GetWindowWMInfo(&mWindow, &mWmInfo);
 
-#if defined(WIN32)
+#if defined(_WIN32)
     wchar_t* path{ nullptr };
     std::ignore = SHGetKnownFolderPath(FOLDERID_ComputerFolder, KF_FLAG_CREATE, nullptr, &path);
     CoTaskMemFree(path);

--- a/OP2-Landlord/FileIo.cpp
+++ b/OP2-Landlord/FileIo.cpp
@@ -37,8 +37,19 @@ namespace
         WideCharToMultiByte(CP_UTF8, 0, &str.at(0), (int)str.size(), &out.at(0), length, nullptr, nullptr);
         return out;
     }
-#endif
 
+
+    std::string peferredSeparatorString()
+    {
+        const std::wstring separatorWchar(1, std::filesystem::path::preferred_separator);
+        return StringFromWString(separatorWchar);
+    }
+#else
+    std::string peferredSeparatorString()
+    {
+        return std::string{1, std::filesystem::path::preferred_separator};
+    }
+#endif
 };
 
 
@@ -53,8 +64,7 @@ FileIo::FileIo(SDL_Window& window) :
     CoTaskMemFree(path);
 #endif
 
-    const std::wstring separatorWchar(1, std::filesystem::path::preferred_separator);
-    mPathSeparator = StringFromWString(separatorWchar);
+    mPathSeparator = peferredSeparatorString();
 }
 
 

--- a/OP2-Landlord/FileIo.cpp
+++ b/OP2-Landlord/FileIo.cpp
@@ -24,7 +24,7 @@ namespace
     };
 
 
-    const std::string StringFromWString(const std::wstring& str)
+    const std::string stringFromWString(const std::wstring& str)
     {
         const auto length = WideCharToMultiByte(CP_UTF8, 0, &str.at(0), (int)str.size(), nullptr, 0, nullptr, nullptr);
 
@@ -42,7 +42,7 @@ namespace
     std::string peferredSeparatorString()
     {
         const std::wstring separatorWchar(1, std::filesystem::path::preferred_separator);
-        return StringFromWString(separatorWchar);
+        return stringFromWString(separatorWchar);
     }
 #else
     std::string peferredSeparatorString()
@@ -173,7 +173,7 @@ bool FileIo::showFileDialog(FileOperation operation, bool pickFolder)
             throw std::runtime_error("Unable to get file name");
         }
 
-        const auto pathStr = StringFromWString(path);
+        const auto pathStr = stringFromWString(path);
         CoTaskMemFree(path);
 
         if (pickFolder)

